spec/system/registration_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe 'User registration', type: :system do
+  before do
+    driven_by(:selenium, using: :chrome)
+  end
+
+  it 'creates and confirms a user' do
+    visit new_user_registration_path
+
+    expect(ActionMailer::Base.deliveries.size).to eq(0)
+
+    sign_up
+
+    user = User.find_by(email: "alen@example.com")
+    expect(user).not_to be_nil
+    expect(user).not_to be_confirmed
+
+    expect(ActionMailer::Base.deliveries.size).to eq(1)
+
+    confirmation_email = ActionMailer::Base.deliveries.last
+    expect(confirmation_email.to).to eq([ "alen@example.com" ])
+    expect(confirmation_email.subject).to eq("Confirmation instructions")
+
+    confirm_email(confirmation_email)
+
+    user.reload
+    expect(user).to be_confirmed
+  end
+
+  def sign_up
+    fill_in "Email", with: "alen@example.com"
```

Comment:
maybe it'd best to pass this email as an argument here, since we have it duplicated above (and we use it for assertions). Something like this:

```ruby
sign_up("alen@example.com")
```